src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -288,4 +288,21 @@ public function unescapeValue($value)
 
         return $value;
     }
+
+    /**
+     * @internal to be removed in 4.0
+     */
+    public static function normalizeName($name)
+    {
+        if (0 === strpos($name, 'env(') && ')' === substr($name, -1) && 'env()' !== $name) {
+            return strtolower($name);
+        }
+
+        $normalizedName = strtolower($name);
+        if ($normalizedName !== (string) $name) {
+            @trigger_error(sprintf('Parameter names will be made case sensitive in Symfony 4.0. Using "%s" instead of "%s" is deprecated since version 3.4.', $name, $normalizedName), E_USER_DEPRECATED);
```

Comment:
Deprecation warnings in Symfony are DX feature, used to point out code that is going to break on update. If we produce false positives, we make it harder for users to detect this kind of issues. It's hard to see, let's say, 2 real issues if they're hidden among 198 false positives.